lib/miq_automation_engine/engine/miq_ae_service_model_base.rb

```diff
@@ -123,9 +125,9 @@ def self.expose(*args)
           return options[:override_return] if options.key?(:override_return)
           reflection = @object.class.reflection_with_virtual(method)
           if reflection && reflection.collection?
-            wrap_results(ret.to_a)
+            options[:association] ? wrap_results(self.class.filter_objects(ret.to_a)) : wrap_results(ret.to_a)
```

Comment:
Rbac only works on active record relations / queries.
The `to_a` converts an active record query into an array, which will need to get translated back to an active record query.

```ruby
# without the to_a:
ems.vms
# with the to_a
Vm.where(:id => ems.vms.map(&:id))
```

The `to_a` forces the second query, which downloads all the records twice. Not good to have at the core of the product.

NOTE: There has been one case where a `distinct` and an `order` clause in the query has conflicted and it was easier to just do the `to_a`. But the order was a subquery function and I've only seen it once.